### PR TITLE
Fix EnableQuery attribute to take into account constant parameterization when PageSize is set (#1523)

### PR DESCRIPTION
**Pull request by adzhiljano**
### Issues

*This pull request fixes issue #1523.*

### Description

*When PageSize is being handled in the EnableQuery attribute it should respect the ODataQuerySettings.EnableConstantParameterization property. Otherwise, there is no way to force a generation of a SQL parameter in the resulting query. Which can result in polluting the SQL Server query plan cache with unwanted query plans (containing constants).*

### Checklist (Uncheck if it is not completed)

- [x] *Test cases added*
- [x] *Build and test with one-click build and test script passed*

### CODE REVIEW

**Comment by biaol-odata** [comment]
Hi @adzhiljano , please run the test shown below and update the publicAPI.bsl per test output. 
```
2018-07-16T08:06:39.1611387Z Failed   Microsoft.AspNet.OData.Test.PublicApi.PublicApiTest.TestPublicApi
```

**Comment by adzhiljano** [comment]
Hey @biaol-odata,

I updated both
`test/UnitTest/Microsoft.AspNet.OData.Test/PublicApi/Microsoft.AspNet.OData.PublicApi.bsl`
and
`test/UnitTest/Microsoft.AspNetCore.OData.Test/PublicApi/Microsoft.AspNetCore.OData.PublicApi.bsl`

Cheers!

File: test/UnitTest/Microsoft.AspNetCore.OData.Test/PublicApi/Microsoft.AspNetCore.OData.PublicApi.bsl [hunk 1]
```diff
@@ -2487,7 +2487,7 @@ public class Microsoft.AspNet.OData.Query.ODataQueryOptions {
 	public bool IsSupportedQueryOption (string queryOptionName)
 	public static bool IsSystemQueryOption (string queryOptionName)
 	public static bool IsSystemQueryOption (string queryOptionName, bool isDollarSignOptional)
-	public static IQueryable`1 LimitResults (IQueryable`1 queryable, int limit, out System.Boolean& resultsLimited)
+	public static IQueryable`1 LimitResults (IQueryable`1 queryable, int limit, bool parameterize, out System.Boolean& resultsLimited)
```
Comment (biaol-odata) [inline test/UnitTest/Microsoft.AspNetCore.OData.Test/PublicApi/Microsoft.AspNetCore.OData.PublicApi.bsl]: We cannot break the public API by adding the bool.
Please consider adding an overload with the original method invoking the new overload with default value.

File: test/UnitTest/Microsoft.AspNetCore.OData.Test/PublicApi/Microsoft.AspNetCore.OData.PublicApi.bsl [hunk 2]
```diff
@@ -2667,9 +2667,9 @@ public class Microsoft.AspNet.OData.Query.TopQueryOption {
 
 public class Microsoft.AspNet.OData.Query.TruncatedCollection`1 : List`1, ICollection`1, IEnumerable`1, IList`1, IReadOnlyCollection`1, IReadOnlyList`1, ICollection, IEnumerable, IList, ICountOptionCollection, ITruncatedCollection {
 	public TruncatedCollection`1 (IEnumerable`1 source, int pageSize)
-	public TruncatedCollection`1 (IQueryable`1 source, int pageSize)
 	public TruncatedCollection`1 (IEnumerable`1 source, int pageSize, System.Nullable`1[[System.Int64]] totalCount)
-	public TruncatedCollection`1 (IQueryable`1 source, int pageSize, System.Nullable`1[[System.Int64]] totalCount)
+	public TruncatedCollection`1 (IQueryable`1 source, int pageSize, bool parameterize)
```
Comment (biaol-odata) [inline test/UnitTest/Microsoft.AspNetCore.OData.Test/PublicApi/Microsoft.AspNetCore.OData.PublicApi.bsl]: same here. We cannot break public API.

**Comment by adzhiljano** [comment]
Fixed the issues as you suggested. I've used `false` as a default value to match the current behaviour. Please take a look at the way I'm getting the `private static readonly MethodInfo _limitResultsGenericMethod` in `ODataQueryOptions` and if that is ok with your conventions. Thanks!

File: src/Microsoft.AspNet.OData.Shared/Query/ODataQueryOptions.cs [hunk 3]
```diff
@@ -436,8 +436,8 @@ internal IQueryable ApplyPaging(IQueryable result, ODataQuerySettings querySetti
             if (pageSize > 0)
             {
                 bool resultsLimited;
-                result = LimitResults(result, pageSize, querySettings.EnableConstantParameterization, out resultsLimited);
-                if (resultsLimited && InternalRequest.RequestUri != null && InternalRequest.RequestUri.IsAbsoluteUri &&
+                result = LimitResults(result, pageSize, out resultsLimited);
```
Comment (mikepizzo) [inline src/Microsoft.AspNet.OData.Shared/Query/ODataQueryOptions.cs]: @KanishManuja-MS:Don't you want to retain the querySettings.EnableConstantParameterization parameter?

**Pull request merged**